src/entity/projectile/WindCharge.php

```diff
@@ -0,0 +1,227 @@
+
+		if(($entity = $source->getDamager()) == null) return;
+		$this->setOwningEntity($entity);
+
+		$this->setMotion($entity->getDirectionVector()->multiply(1.5));
+	}
+
+	protected function onHitEntity(Entity $entityHit, RayTraceResult $hitResult) : void{
+		if($this->getOwningEntity() === null)$ev = new EntityDamageByEntityEvent($this, $entityHit, EntityDamageEvent::CAUSE_PROJECTILE, $this->damage);
+		else $ev = new EntityDamageByChildEntityEvent($this->getOwningEntity(), $this, $entityHit, EntityDamageEvent::CAUSE_PROJECTILE, $this->damage);
+		$entityHit->attack($ev);
+
+		$this->flagForDespawn();
+	}
+
+	protected function onHitBlock(Block $blockHit, RayTraceResult $hitResult) : void{
+		parent::onHitBlock($blockHit, $hitResult);
+		$bound = $this->getBound($blockHit->getPosition(), $this->radius - 1);
+
+		for($x = $bound->minX; $x <= $bound->maxX; $x++) {
+			for($y = $bound->minY; $y <= $bound->maxY; $y++) {
+				for($z = $bound->minZ; $z <= $bound->maxZ; $z++) {
+					$affectedBlock = $this->getWorld()->getBlockAt((int) floor($x), (int) floor($y), (int) floor($z));
+
+					$this->checkAffect($affectedBlock);
+				}
+			}
+		}
+	}
+
+	protected function entityBaseTick(int $tickDiff = 1) : bool{
+		parent::entityBaseTick($tickDiff);
+
+		if($this->createdAt + $this->lifetime <= time()) $this->flagForDespawn();
+
+		return true;
+	}
+
+	protected function onHit(ProjectileHitEvent $event) : void{
+		$source = $this->getLocation();
+
+		$this->getWorld()->addSound($event->getRayTraceResult()->getHitVector(), new WindChargeBurstSound());
+		$this->getWorld()->addParticle($source, new WindExplosionParticle());
+
+		$list = $source->getWorld()->getNearbyEntities($this->getBound($source, $this->radius));
+
+		foreach($list as $entity){
+			if($entity == $this) continue;
+
+			$entityPos = $entity->getPosition();
+			$distance = $entityPos->distance($source) / $this->radius;
+			$motion = $entityPos->subtractVector($source)->normalize();
+
+			($entity->isUnderwater()) ? $exposure = 0.5 : $exposure = 1.5;
+			$impact = (1 - $distance) * $exposure;
+
+			if ($impact <= 0) continue;
+
+			if (round($entityPos->getX(), 1) == round($source->getX(), 1) && round($entityPos->getZ(), 1) == round($source->getZ(), 1)) {
+				$entity->setMotion($entity->getMotion()->add(0, 0.5 * $exposure, 0));
+
+				return;
+			}
+
+			$entity->setMotion($entity->getMotion()->add(0, $impact * 0.4, 0)->addVector($motion->multiply($impact)));
+		}
+	}
+
+	private function getBound(Vector3 $source, float $radius) : AxisAlignedBB {
+		$minX = (int) floor($source->x - $radius - 1);
+		$maxX = (int) ceil($source->x + $radius + 1);
+		$minY = (int) floor($source->y - $radius - 1);
+		$maxY = (int) ceil($source->y + $radius + 1);
+		$minZ = (int) floor($source->z - $radius - 1);
+		$maxZ = (int) ceil($source->z + $radius + 1);
+
+		return new AxisAlignedBB($minX, $minY, $minZ, $maxX, $maxY, $maxZ);
+	}
+
+	private function checkAffect(Block $affectedBlock) : void {
```

Comment:
Would be more consistent create a new method on the Block class and each block will decide it's own behavour